docs/14-batch-shipyard-configuration-jobs.md

```diff
@@ -1038,14 +1038,19 @@ property are:
         * `file_mode` if the file mode to set for the file on the compute node.
           This is optional.
     * `pre_execution_command` is a command that is run only on the master node
-      of this multi-instance task prior to the application command. This
-      command must not block and must exit successfully for the multi-instance
-      task to proceed. This command can be used to populate environment
-      variables required to run the application command. This is optional and
-      may be null. Note that this command cannot be used with Docker images.
+      of this multi-instance task prior to the application command. For
+      Docker containers, this command is executed in the container just before
+      the application command is executed. For Singularity containers, this
+      command is executed on the host just before the application command is
+      executed. This command must not block and must exit successfully for the
+      multi-instance task to proceed. This command can be used to populate
+      environment variables required to run the application command. This is
+      optional and may be null.
     * (required if using MPI) `mpi` contains the following members:
         * (required) `runtime` is the runtime that should be used. Valid
-          values are `intelmpi` and `openmpi`.
+          values are `intelmpi`, `mpich`, and `openmpi`. With Docker
+          containers, it is the user's responsability to provide a container
+          image that has the specified runtime installed.  
```

Comment:
For Singularity containers, the specified runtime must be installed and loaded on the host compute node.